src/presentation/api/auth.py

```
@@ -103,4 +103,23 @@ async def get_current_user(
     
     # 사용자 조회 로직이 필요하다면 여기에 추가
     user = User(id=int(user_id), kakao_id=payload.get("kakao_id"), nickname=payload.get("nickname"))
-    return user
\ No newline at end of file
+    return user
+
+
+@router.delete("/withdraw")
+@inject
+def withdraw_user(
+    current_user: User = Depends(get_current_user),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)